x-pack/plugins/file_upload/public/components/json_upload_and_parse.js

```diff
@@ -87,23 +89,39 @@ export class JsonUploadAndParse extends Component {
     });
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    if (!_.isEqual(prevState.parsedFile, this.state.parsedFile)) {
```

Comment:
Yes, this could be a very expensive equality check if the file is large - you could be testing 50MBs for equality! The guard ensuring `indexTypes` gets updated when the file content changes. So instead of comparing the entire file, the refactored logic just checks if `indexTypes` changed and updates state if it has.